### PR TITLE
Fix routing/page organization

### PR DESCRIPTION
**Pull request by tsumori91**
### What this PR does (required):
- Re-routes all profile related pages so they are a child of one new page.
- In the new page fetching profile data from the database to pass down to children via props
- Now creates a new profile if one is not found in the database when Editing profile (I wasn't sure how we want to deal with this but the framework is setup in editProfile page for now)

### Screenshots / Videos (required):
- N/A

### Any issues with the current functionality (optional):
- Need to confirm that the pages properly get profile data before loading is set to false (I did a bit, but am still not confident)

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -31,13 +26,8 @@ function App(): JSX.Element {
...
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
-                <Route exact path="/editprofile" component={EditProfile} />
-                <Route exact path="/payment" component={Payment} />
-                <Route exact path="/availability" component={Availability} />
-                <Route exact path="/security" component={Security} />
-                <Route exact path="/settings" component={Settings} />
-                <Route exact path="/profile/pics" component={ProfilePhoto} />
                 <Route exact path="/profile" component={SitterProfile} />
+                <ProtectedRoutes path="/manageprofile" component={ManageProfile} />
```
Comment (rajivtitus) [inline client/src/App.tsx:30]: Please use hyphenated case for route names 
Eg: `manage-profile`

File: client/src/components/Sidebar/SideBar.tsx [hunk 2]
```diff
@@ -20,22 +20,22 @@ const Sidebar = (): JSX.Element => {
     >
       <div className={classes.toolbar} />
       <List>
-        <ListItem button component={Link} to="/editprofile">
+        <ListItem button component={Link} to="/manageprofile/editprofile">
```
Comment (rajivtitus) [inline client/src/components/Sidebar/SideBar.tsx:23]: Please change route names to hyphenated case. Ditto throughout :)

Eg: `/manage-profile/edit-profile`

File: client/src/pages/ManageProfile/ManageProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+export default function ManageProfile(): JSX.Element {
+  const classes = useStyles();
+  const [loading, setLoading] = useState(true);
+
+  const location = useLocation();
+  const userData = useAuth();
+  const userId = userData.loggedInUser?.id;
+  const [userProfile, setUserProfile] = useState<Profile>();
+
+  const handleRoute = (path: string) => {
```
Comment (rajivtitus) [inline client/src/pages/ManageProfile/ManageProfile.tsx:26]: 👍🏽

File: client/src/pages/ManageProfile/ManageProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  }, [userId]);
+  if (userId === undefined) return <CircularProgress />;
+  if (loading) return <CircularProgress />;
+  return (
+    <Grid container>
+      <CssBaseline />
+      <Grid item xs={2}>
+        <SideBar />
+      </Grid>
+      <Grid item xs={10} style={{ backgroundColor: '#FAFAFA' }}>
```
Comment (rajivtitus) [inline client/src/pages/ManageProfile/ManageProfile.tsx:72]: Please avoid the use of inline-styles

File: client/src/pages/ManageProfile/ManageProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+      default:
+        return <EditProfile />;
+    }
+  };
+  useEffect(() => {
+    setLoading(true);
+    async function getProfileData() {
+      if (userId)
+        return await getUserProfile(userId).then((data) => {
```
Comment (rajivtitus) [inline client/src/pages/ManageProfile/ManageProfile.tsx:54]: Maybe you can get the user's profile and store it in the auth context? That way profile information is accessible in other components as well (Navbar for eg).

File: client/src/pages/ManageProfile/ManageProfile.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+          if (!data?.success) {
+            setLoading(false);
+          } else setUserProfile(data.success.profile);
+          setLoading(false);
+        });
```
Comment (rajivtitus) [inline client/src/pages/ManageProfile/ManageProfile.tsx:59]: You can set loading to false just once outside of the if/else clause

File: client/src/App.tsx (hunk 1 above)
Comment (tsumori91) [reply client/src/App.tsx:30]: Didn't realize that was the convention, thank you!

**Review by rajivtitus — APPROVED** [review]
As we discussed that the profile will be factored into the context at a later stage, I am approving this PR. 

Please make necessary commits to address comments (if you've not already done so)
Thank you!

**Pull request merged**